### PR TITLE
[Sprint 3] List notification ui

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Link redmine: https://dev.framgia.com/redmine/issues/114168
Task: List notification
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
+                        R.layout.item_list_notification, parent, false);
+        return new NotificationPageAdapter.ItemViewHolder(itemListNotificationBinding,
+                mItemClickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        ItemViewHolder itemViewHolder = (ItemViewHolder) holder;
+        itemViewHolder.bind(mNotifications.get(position));
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java]: change to
```java
if (holder instanceof ItemViewHolder) {
ItemViewHolder itemViewHolder = (ItemViewHolder) holder;
...
}

File: app/src/main/java/com/framgia/forder/screen/notification/NotificationPageViewModel.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    public String getNotificationTitle() {
+        return mNotification != null ? mNotification.getTitle() : "";
+    }
+
+    public String getNotificationTime() {
+        return mNotification != null ? mNotification.getTime() : "";
+    }
+
+    public LayoutManagers.LayoutManagerFactory getLayoutManager() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/notification/NotificationPageViewModel.java]: remove this method

File: app/src/main/res/layout/fragment_list_notification.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            name="viewModel"
+            type="com.framgia.forder.screen.notification.NotificationPageViewModel"
+            />
+
+    </data>
+
+    <android.support.v7.widget.RecyclerView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:layoutManager="@{viewModel.layoutManager}"
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_list_notification.xml]: change to
```xml
app:layoutManager="LinearLayoutManager"
```

File: app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+public class NotificationPageAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
+
+    private OnRecyclerViewItemClickListener<Object> mItemClickListener;
+    private List<Notification> mNotifications;
+
+    NotificationPageAdapter(@NonNull Context context, List<Notification> notifications) {
+        super(context);
+        mNotifications = new ArrayList<>();
+        mNotifications.addAll(notifications);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java]: if notifications ==null appear crash

File: app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import com.framgia.forder.databinding.ItemListNotificationBinding;
+import com.framgia.forder.screen.BaseRecyclerViewAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ASUS on 24-04-2017.
+ */
+
+public class NotificationPageAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java]: pls change 
extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> 
to 
 -> extends BaseRecyclerViewAdapter<NotificationPageAdapter.ItemViewHolder> {

and 

private static class ItemViewHolder

-> static class ItemViewHolder

Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/forder/screen/notification/NotificationPageAdapter.java]: dạ ok

**Pull request merged**